matching-notification-api/src/main/java/org/phenotips/matchingnotification/match/internal/DefaultPatientInMatch.java

```diff
@@ -0,0 +1,248 @@
+
+    private static final PatientRepository PATIENT_REPOSITORY;
+
+    private static final String GENES = "genes";
+
+    private static final String PHENOTYPES = "phenotypes";
+
+    private String patientId;
+
+    private String serverId;
+
+    private String href;
+
+    private Set<String> genes;
+
+    private PhenotypesMap phenotypes;
+
+    static {
+        PatientMatchNotifier notifier = null;
+        PatientGenotypeManager pgm = null;
+        PatientRepository patientRepository = null;
+        try {
+            ComponentManager ccm = ComponentManagerRegistry.getContextComponentManager();
+            notifier = ccm.getInstance(PatientMatchNotifier.class);
+            pgm = ccm.getInstance(PatientGenotypeManager.class);
+            patientRepository = ccm.getInstance(PatientRepository.class);
+        } catch (ComponentLookupException e) {
+            LOGGER.error("Error loading static components: {}", e.getMessage(), e);
+        }
+        NOTIFIER = notifier;
+        PATIENT_GENOTYPE_MANAGER = pgm;
+        PATIENT_REPOSITORY = patientRepository;
+    }
+
+    /**
+     * Builds an object from PatientMatch and Patient objects. This is used when a PatientMatch object
+     * is built the first time.
+     *
+     * @param match the match that contains the patient this object represents
+     * @param patient the patient that this object represents
+     * @param serverId id of server where patient is found
+     */
+    public DefaultPatientInMatch(PatientMatch match, Patient patient, String serverId)
+    {
+        this.patientId = patient.getId();
+        this.serverId = serverId;
+        this.href = match.getHref();
+
+        this.readDetails(patient);
+    }
+
+    /**
+     * Builds an object for a patient from a PatientMatch and the result of a
+     * call to getDetailsColumn. This is used when a PatientMatch is retrieved
+     * from the DB.
+     *
+     * @param match
+     *            the match that contains the patient this object represents
+     * @param patientId
+     *            the id of the patient that this object represents
+     * @param serverId
+     *            id of server where patient is found
+     * @param patientDetails
+     *            the result of a previous call to getDetailsColumn.
+     */
+    public DefaultPatientInMatch(PatientMatch match, String patientId, String serverId, String patientDetails)
+    {
+        this.patientId = patientId;
+        this.serverId = serverId;
+        this.href = match.getHref();
+
+        this.rebuildDetails(patientDetails);
+    }
+
+    @Override
+    public JSONObject toJSON()
+    {
+        JSONObject json = new JSONObject();
+        json.put("patientId", this.getPatientId());
+        json.put("serverId", this.getServerId());
+        json.put(GENES, this.genes);
+        json.put(PHENOTYPES, this.phenotypes.toJSON());
+        json.put("emails", this.getEmails());
```

Comment:
Is there a reason this explicitly adds genes and phenotypes, instead of merging in the getDetailsColumn data?